tensorflow/python/eager/forwardprop.py

```diff
@@ -140,6 +142,44 @@ def _jvp_helper(op_name, attr_tuple, inputs, outputs, tangents):
     return output_tangents
 
 
+def _jvp_helper_wrapper(
+  op_name, attr_tuple, inputs, outputs, tangents, use_batch 
+):
+  """Computes a batch of Jacobian-vector product for an op.
+
+  Args:
+    op_name: A string, the type of operation being executed.
+    attr_tuple: Attributes of the operation.
+    inputs: A flat list of input Tensors to the operation.
+    outputs: A flat list of output Tensors from the operation.
+    tangents: A flat list of Tensors, compatible with shape `[None] + input_shape`.
+    use_batch: A bool, True to vetorize over batch of tangents 
+      of shape `[None] + input_shape`
+
+  Returns:
+    A flat list of tangents compatible with `outputs` or `[None] + output_shape`.
+  
+  Raises:
+    ValueError: if tangent shapes are not compatible with input shapes.
+  """
+  if use_batch:
+    for primal, tangent in zip(inputs, tangents):
+      if tangent.shape.is_compatible_with(primal.shape): 
```

Comment:
Shouldn't it be "if not tangent.shape.is_compatible_with([None] + primal.shape):"?

You might want to use assertRaisesRegexp; maybe that test was throwing a different ValueError?

Reply:
My bad, the ValueError was for no same nested structure.